plane2/src/drone/backend_manager.rs

```diff
@@ -59,7 +81,10 @@ pub struct BackendManager {
     executor_config: ExecutorConfig,
 
     /// Function to call when the state changes.
-    state_callback: StateCallback,
+    state_callback: Callback<BackendState>,
+
+    /// Function to call when metrics produced.
+    metrics_callback: Callback<Stats>,
```

Comment:
Given that this callback will likely just pass the data over the socket, should we just use a `TypedSender` for this? This PR gave me an idea for improving the ergonomics of `TypedSender` for this purpose, which I opened here: https://github.com/drifting-in-space/plane/pull/508